File: python/src/Creational/Prototype/AbstractFork.py


#//AbstractFork - Two of Two Prototypes

import copy

class AbstractFork:

	def __init__(self):
		raise Exception("Can't instantiate abstract class, AbstractFork")

	def setForkName(self,forkName) :
		self.forkName = forkName

	def getForkName(self) : 
		return self.forkName

	def clone(self) :
		return copy.deepcopy(self)

File: ruby/src/Creational/Prototype/AbstractFork.rb


#//AbstractFork - Two of Two Prototypes

class AbstractFork

	attr_accessor :forkName

	def initialize()
		raise "Can't instantiate abstract class, AbstractFork"
	end

	def setForkName(forkName)
 		@forkName = forkName
	end

	def getForkName()
 		return @forkName
	end

end

